tls/s2n_config.c

```diff
@@ -127,6 +127,8 @@ static int s2n_config_init(struct s2n_config *config)
     config->cert_tiebreak_cb = NULL;
 
     GUARD(s2n_config_setup_default(config));
+
+    config->minimum_protocol_version = config->cipher_preferences->minimum_protocol_version;
```

Comment:
TLS1.3 is still a pre-release feature, so we shouldn't allow the minimum version to be set to TLS1.3 unless `s2n_is_tls13_enabled()` is true.